client/app/components/calendar/event/DeleteButton.jsx

```diff
@@ -0,0 +1,38 @@
+import React, { PropTypes } from 'react';
+import axios from 'axios';
+import './event.scss';
+
+export default class DeleteButton extends React.Component {
+  static propTypes = {
+    id: PropTypes.string.isRequired
+  };
+
+  _path() {
+    return `/events/${this.props.id}`;
+  }
+
+  _processDelete(event) {
+    event.preventDefault();
+    let token = document.querySelector('meta[name="csrf-token"]').content;
+
+    axios({
+      method: 'DELETE',
+      url: this._path(),
+      headers: {
+        'X-CSRF-Token': token
```

Comment:
If you look into Refresh button #68 pull request I added an EventSource file.
I think it's a good convention to just provide a simple API to components for requests.
Also it would be cleaner 'cuz then there would be a single place where you're handling the token.